packages/aws-cdk/lib/util/please-hold.ts

```diff
@@ -0,0 +1,25 @@
+import { warning } from "../logging";
+
+/**
+ * Print a message to the logger in case the operation takes a long time
+ */
+export class PleaseHold {
+  private handle?: NodeJS.Timer;
+
+  constructor(private readonly message: string, private readonly timeoutSec = 10) {
+  }
+
+  public start() {
+    this.handle = setTimeout(this.printMessage.bind(this), this.timeoutSec * 1000);
```

Comment:
This doesn't print the message out when the operation starts, only if it takes more than `timeoutSec` seconds. That could be a heck of a long time... Maybe display the message once first, then wait?

Reply:
But that's the point of the class. If everything proceeds quickly, no need to be verbose. But if things take a long time, then tell the user what they're waiting for.